lib/Config.js

```diff
@@ -488,16 +472,15 @@ class Config {
                 'bad environment variable: S3DATA environment variable ' +
                 'should be one of mem/file/scality/multiple'
             );
-            if (process.env.S3DATA === 'multiple') {
-                assert(config.locationConstraints !== undefined,
-                    'for multiple data backends, locationConstraints ' +
-                    'must be set'
-                );
-            }
             data = process.env.S3DATA;
         }
-        if (process.env.S3SPROXYD) {
-            data = process.env.S3SPROXYD;
+        if (data === 'scality' || data === 'multiple') {
+            data = 'multiple';
+            assert(this.locationConstraints !== undefined &&
```

Comment:
don't you do this check regardless of the data value?  they should have locationConstraints and restEndpoints no matter what